src/main/java/com/hotels/beeju/BeejuJUnitRule.java

```diff
@@ -33,58 +31,32 @@
  */
 abstract class BeejuJUnitRule extends ExternalResource {
 
-  @VisibleForTesting
-  final TemporaryFolder temporaryFolder = new TemporaryFolder();
-  private File metastoreLocation;
-  public BeejuCore core;
+  protected BeejuCore core;
 
-  public BeejuJUnitRule(String databaseName, Map<String, String> configuration) {
-    core = new BeejuCore(databaseName,configuration);
+  BeejuJUnitRule(String databaseName, Map<String, String> configuration) {
+    core = new BeejuCore(databaseName, configuration);
   }
 
   /**
-   * Initialise the warehouse path.
-   * <p>
    * This method can be overridden to provide additional initialisations.
    * </p>
-   *
-   * @throws Throwable If the initialisation fails.
    */
-  protected void init() throws Throwable {
-    metastoreLocation = temporaryFolder.newFolder("metastore");
-    core.setHiveVar(HiveConf.ConfVars.METASTOREWAREHOUSE, metastoreLocation.getAbsolutePath());
+  protected void init() throws Exception {
   }
 
-  /**
-   * Implement this method to prepare the rule for a new test.
-   * <p>
-   * This is called after the rule is initialised.
-   * </p>
-   *
-   * @throws Throwable If the rule cannot be prepared for a new test.
-   */
-  protected abstract void beforeTest() throws Throwable;
-
   @Override
   protected void before() throws Throwable {
-    temporaryFolder.create();
     init();
-    beforeTest();
     createDatabase(databaseName());
   }
 
-  /**
-   * Implement method to release any resources used by the rule.
-   * <p>
-   * This method is called before the warehouse directory is deleted.
-   * </p>
-   */
-  protected abstract void afterTest();
-
   @Override
   protected void after() {
-    afterTest();
-    temporaryFolder.delete();
+    try {
+      core.cleanUp();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
Since we're now on Java8 we can use https://docs.oracle.com/javase/8/docs/api/java/io/UncheckedIOException.html instead of RTE.